lib/util/Bucket.js

```diff
@@ -13,23 +13,30 @@ class Bucket {
     * Construct a Bucket
     * @arg {Number} tokenLimit The max number of tokens the bucket can consume per interval
     * @arg {Number} interval How long (in ms) to wait between clearing used tokens
+    * @arg {Number} reserved How many tokens to reserve for priority operations
     * @arg {Object} [latencyRef] An object
     * @arg {Number} latencyRef.latency Interval between consuming tokens
     */
-    constructor(tokenLimit, interval, latencyRef = {latency: 0}) {
+    constructor(tokenLimit, interval, reserved = 0, latencyRef = {latency: 0}) {
```

Comment:
Not a fan of adding it like this as it breaks things... Could use an options object as 4th param

Reply:
Strikes me as a bit odd to do it like that. Alternatively, we could merge `latencyRef` and `reserved` into a single options object. I don't even see anywhere in the lib where we explicitly define `latencyRef`, though search could be failing me.